src/net/ip6/mld.cpp

```diff
@@ -0,0 +1,241 @@
+// This file is a part of the IncludeOS unikernel - www.includeos.org
+//
+// Copyright 2015 Oslo and Akershus University College of Applied Sciences
+// and Alfred Bratterud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//#define MLD_DEBUG 1
+#ifdef MLD_DEBUG
+#define PRINT(fmt, ...) printf(fmt, ##__VA_ARGS__)
+#else
+#define PRINT(fmt, ...) /* fmt */
+#endif
+#include <vector>
+#include <net/inet>
+#include <net/ip6/mld.hpp>
+#include <net/ip6/packet_mld.hpp>
+#include <net/ip6/icmp6.hpp>
+#include <statman>
+
+namespace net
+{
+  Mld::Mld(Stack& inet) noexcept:
```

Comment:
Minor stuff, but would you mind typing these constructors differently?
```
Mld::Mld(Stack& inet) noexcept
  : inet_{inet},
  host_{*this},
  router_{*this}
```
Is the most used one, but I guess everything works as long as it's indentation on the member initialization.